### PR TITLE
Got workflow working with act

### LINKED ISSUE #85: Unit testing suite
## Check list

- [X] I have read through the [README](https://github.com/wfxr/minimap.vim/blob/master/README.md) (especially F.A.Q section)
- [X] I have searched through the existing issues

## Related issues
GH-81

## Solution you'd like

Now that there are multiple optional features, it would be good to hook in some unit tests to make sure changes to one feature don't affect another feature. 

I'll implement the request, just putting it here for visibility and feedback.

I was looking into https://github.com/dhruvasagar/vim-testify, but I'm not married to anything.

## Checklist
- [x] Add tests
- [x] Update documentation on how to run the unit tests
- [ ] Integrate them into a git workflow (Install act to test workflow locally)

### PR DESCRIPTION
**Pull request by ZNielsen**
## Check list

- [x] I have read through the [README](https://github.com/wfxr/minimap.vim/blob/master/README.md) (especially F.A.Q section)
- [x] I have searched through the existing issues or pull requests
- [x] I have verified all existing unit tests pass (See the README on how to run)
- [x] I have added new tests if appropriate
- [x] I have performed a self-review of my code and commented hard-to-understand areas
- [x] I have made corresponding changes to the documentation (when necessary)

## Description

Closes #85 
- Adds unit test running to the workflow
- Removed any case-based search criteria from the tests since it depends on user settings
- Breaks up the search tests so it's easier to see what failed

## Type of change

- [ ] Bug fix
- [ ] New feature
- [x] Improvement of existing features
- [x] Refactor
- [ ] Breaking change
- [ ] Documentation change
- [x] CI / CD

## Test environment

- OS
    - [ ] Linux
    - [x] Mac OS X
    - [ ] Windows
    - [ ] Others:
- Vim
    - [x] Neovim: 0.5
    - [ ] Vim: <Version>

### CODE REVIEW

**Comment by ZNielsen** [comment]
Tested the workflow with act, but it looks like there's more work to do to get it working with the actual workflow.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -12,6 +12,30 @@ jobs:
...
+          neovim: true
+      - name: Ensure plugins directory exists
+        run: |
+          mkdir -p $HOME/.config/nvim/plugins
+      - name: Create a fake code-minimap executable
+        # Necessary for minimap.vim to start up
+        run: |
+          mkdir -p "$HOME/bin"
+          ln -s /bin/echo "$HOME/bin/code-minimap"
+          echo "$HOME/bin" >> $GITHUB_PATH
```
Comment (wfxr) [inline .github/workflows/ci.yml:35]: What does this line do? Equivalent to append `$HOME/bin` to the `PATH` environment variable?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -12,6 +12,30 @@ jobs:
...
+      - name: Checkout Testify
+        run: |
+          git clone --depth 1 https://github.com/dhruvasagar/vim-testify $HOME/.config/nvim/plugins/vim-testify
+      - name: Run Unit Tests
+        run: |
+          nvim -c "set rtp+=$HOME/.config/nvim/plugins/vim-testify" -c "set rtp+=./" -S $HOME/.config/nvim/plugins/vim-testify/plugin/testify.vim -S ./plugin/minimap.vim +TestifySuite
```
Comment (wfxr) [inline .github/workflows/ci.yml:41]: Wow, this way is much simpler and better than I thought :+1: !

**Review by wfxr — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ZNielsen) [reply .github/workflows/ci.yml:35]: Yes, it updates the `PATH`. I think `GITHUB_PATH` is a bit different in that it is available for every subsequent step without needing to re-source anything. When I looked up how to update the `PATH` in the docs, this was the recommended way so I just went with it.

Comment (wfxr) [reply .github/workflows/ci.yml:35]: Thank you for the explanation @ZNielsen

**Pull request merged**